include/text_view_detail/otext_iterator.hpp

```diff
@@ -76,42 +95,35 @@ class otext_cursor
         current = tmp;
     }
 
-private:
-    iterator_type current;
-};
-
-
-template<TextEncoding ET, CodeUnitOutputIterator<code_unit_type_t<ET>> CUIT>
-class otext_iterator_mixin
-    : protected ranges::basic_mixin<otext_cursor<ET, CUIT>>
-{
-    using iterator_type = CUIT;
-    using cursor_type = otext_cursor<ET, CUIT>;
-    using base_type = ranges::basic_mixin<cursor_type>;
-
-public:
-    using encoding_type = ET;
-    using state_type = typename ET::state_type;
-    using state_transition_type = typename ET::state_transition_type;
-
-    otext_iterator_mixin() = default;
-
-    otext_iterator_mixin(
-        state_type state,
-        iterator_type current)
-    :
-        base_type{cursor_type{std::move(state), std::move(current)}}
+    void next() noexcept
     {}
 
-    using base_type::base_type;
-
-    const state_type& state() const noexcept {
-        return this->get().state();
+    auto post_increment() noexcept {
+        class proxy
+        {
+        public:
+            proxy(otext_cursor& self) noexcept
+                : self_(self)
+            {}
+            proxy& operator*() noexcept {
+                return *this;
+            }
+            proxy& operator=(const state_transition_type &stt) {
+                self_.write(stt);
+                return *this;
+            }
+            proxy& operator=(const character_type_t<encoding_type> &value) {
+                self_.write(value);
+                return *this;
+            }
+        private:
+            otext_cursor& self_;
+        };
+        return proxy{*this};
```

Comment:
This change actually resulted in otext_iterator no longer being an output iterator since the returned proxy type is not convertible to the iterator type.  I addressed this in commit d162dc8099de916a25c992993d4646a6039e66f0.  It seems that ranges::OutputIterator is missing the requirement for post-increment to return a type convertible to the iterator type.